src/videojs-contrib-hls.js

```diff
@@ -216,18 +216,31 @@ Hls.canPlaySource = function() {
  */
 Hls.supportsNativeHls = (function() {
   let video = document.createElement('video');
-  let xMpegUrl;
-  let vndMpeg;
 
   // native HLS is definitely not supported if HTML5 video isn't
   if (!videojs.getComponent('Html5').isSupported()) {
     return false;
   }
 
-  xMpegUrl = video.canPlayType('application/x-mpegURL');
-  vndMpeg = video.canPlayType('application/vnd.apple.mpegURL');
-  return (/probably|maybe/).test(xMpegUrl) ||
-    (/probably|maybe/).test(vndMpeg);
+  // HLS manifests can go by many mime-types
+  let canPlay = [
+    // Apple santioned
+    video.canPlayType('application/vnd.apple.mpegURL'),
+    // Apple sanctioned for backwards compatibility
+    video.canPlayType('audio/mpegURL'),
+    // Very common
+    video.canPlayType('audio/x-mpegURL'),
+    // Very common
+    video.canPlayType('application/x-mpegURL'),
+    // Included for completeness
+    video.canPlayType('video/x-mpegURL'),
+    video.canPlayType('video/mpegURL'),
+    video.canPlayType('application/mpegURL')
+  ];
+
+  return canPlay.some(function(canItPlay) {
+    return (/maybe|probably/i).test(canItPlay);
```

Comment:
Since we just want to test that they are not an empty string, would it be simpler to remove the array and the some call and just do a return with ors between each?

return video.canPlayType... || video.canPlayType... || ...

Or are we protecting against other potential values of canPlayType?